axelrod/strategies/mindreader.py

```diff
@@ -46,15 +46,25 @@ class MindReader(Player):
     """A player that looks ahead at what the opponent will do and decides what to do."""
 
     name = 'Mind Reader'
-    stochastic = True # Don't cache me
+    behaviour = {
+        'memory_depth': -10,
+        'stochastic': False,
```

Comment:
I think this one is stochastic, since the simulated play could be stochastic. MindReader alters opponent state rather than the source (so those two booleans are flipped).

I'm not sure if it actually inspects the source -- it doesn't explicitly read the source with `inspect`, but it does use the source to simulate. Let's be sure that the docs explain that both behaviors are included.